### PR TITLE
Created FileGateway and added delivery mechanism

### PR DESCRIPTION
**Pull request by georgeschena**
Moved away from saving in local memory to using a persistence way of storing the marks

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,8 @@ source "https://rubygems.org"
 
 git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
 
-gem "rails"
+#gem "rails"
```
Comment (naliwajek) [inline Gemfile]: Do not live commented out code. For history, we have git.

File: lib/file_mark_gateway.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    File.open('/tmp/test', 'w') do |file|
+      file.write(mark.type_of_mark + " " + (mark.cell_number).to_s + "\n")
+    end
+  end
+
+  def delete_all
+    File.unlink('/tmp/test')
+  rescue
+  end
+end
```
Comment (naliwajek) [inline lib/file_mark_gateway.rb]: You need to set up your editor to add a newline at the end of files.

File: lib/web_app.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby -I ../lib -I lib
```
Comment (naliwajek) [inline lib/web_app.rb]: Interesting. Do you plan to replace it with some other loading in the future?

File: spec/unit/file_mark_gateway_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+describe FileMarkGateway do
+  before { FileMarkGateway.new.delete_all }
+
```
Comment (naliwajek) [inline spec/unit/file_mark_gateway_spec.rb]: I believe this is going to be cool: http://www.betterspecs.org/#subject

File: lib/file_mark_gateway.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    contents = File.read('/tmp/test')
+    contents = contents.split(" ")
+    [
+      Mark.new(contents[0], contents[1].to_i)
+    ]
+  end
+
+  def mark_properties=(mark)
+    File.open('/tmp/test', 'w') do |file|
+      file.write(mark.type_of_mark + " " + (mark.cell_number).to_s + "\n")
```
Comment (craigjbass) [inline lib/file_mark_gateway.rb]: Nice

File: lib/web_app.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @mark_properties
+  end
+
+  def mark_properties=(mark_properties)
+    @mark_properties.push(mark_properties)
+  end
+end
+
+
+get '/' do
```
Comment (craigjbass) [inline lib/web_app.rb]: This PR also adds your delivery mechanism code. The PR title lies :cold-sweat:

File: spec/unit/file_mark_gateway_spec.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  it 'returns one mark X from the array' do
+    gateway = FileMarkGateway.new
+    gateway.mark_properties = Mark.new('X', 2)
+    expect(gateway.mark_properties[0].type_of_mark).to eq('X')
+    expect(gateway.mark_properties[0].cell_number).to eq(2)
+  end
+
+  it 'returns two marks from the array' do
```
Comment (craigjbass) [inline spec/unit/file_mark_gateway_spec.rb]: master should pass all tests.

**Pull request merged**